src/shapes/Object/Object.ts

```diff
@@ -845,7 +842,7 @@ export class FabricObject<
       this.paintFirst === 'stroke' &&
       this.hasFill() &&
       this.hasStroke() &&
-      typeof this.shadow === 'object'
+      this.shadow !== undefined
```

Comment:
This is the fix if I understand correctly

Reply:
To repsect undefined, null, fabric.Object:
```suggestion
      !!this.shadow
```

Reply:
So you're thinking even though we typed it to object or undefined, null might sneak in somehow?

Reply:
no
I am pointing this out to stress that the fix is not blocked by the undefined change

Reply:
and safer anyways

Reply:
yes i agree, !!this.shadow accomodate everything for now and does not require us to do a large change like this.
I would prefer to change the types in a different PR.

Reply:
For the types change, are you suggesting changing more types from null? Just wanted to see the idea for the other PR

Reply:
Understood. We just don't want to touch as many times.